Add application upgrade docs (#237)
pluralsh/documentation: pages/getting-started/understanding-upgrades.md

An app can have any number of release channels which are basically named tags for specific versions of an app. In general these will be:

- latest - always the last version managed by our system
- stable - a channel we frequently maintain based on integration tests we've written for apps
- <vsn> - a channel for specific app versions

You can toggle any release channel in the `/clusters` page of https://app.plural.sh. This will often result in a lot of your installation being rewired, which will then flush to your console. Here's an example of the upgrade channels for an airbyte installation:

![](/assets/upgrade-channel.png)

## The Upgrade Queue

When a new version of a helm chart or terraform module is pushed into a release channel, our api will update your installation for you and then push an upgrade into your upgrade queue for the console to apply for you. This is an ordered list which is meant to be dependency ordered to properly guide the console to safely apply an upgrade.

That said the source of truth of what you install is your terraform/helm installation in our api and this is really a sequence of commands the console is meant to execute on your behalf.

> You can see an upgrade queue for a cluster in the clusters tab in app.plural.sh at any point in time as well.

Reviewer: ```suggestion
You can see an upgrade queue for a cluster in the [clusters tab](https://app.plural.sh/overview/clusters) in app.plural.sh at any point in time as well.
```

## Cluster Promotions

Cluster promotions are a mechanism to control all upgrades to a given cluster to allow for dev -> staging -> prod separation. It's a useful risk management mechanism especially if you're using apps like airflow that has custom code you inject in the airflow instance that needs to be tested for yourself alongside any given version.

> You must be a Pro or Enterprise user of Plural to access promotions, but if you are, it's a pretty simple process to setup new promotions, this video will give you a walkthrough:

Reviewer: ```suggestion
You must be a Pro or Enterprise user of Plural to access promotions, but if you are, it’s a pretty simple process to setup new promotions, this video will give you a walkthrough:
```

{% embed url="https://www.youtube.com/watch?v=ZGXZmqelHB0&ab_channel=Plural" aspectRatio="16 / 9" /%}

The way the promotion mechanism works is it defers all upgrades for the cluster up to a given point in time. When you promote from a source cluster, it will move that waterline to whatever the source clusters waterline was (or the current time if it doesn't have a source cluster of its own). This allows you to set up long promotion chains like `dev -> staging -> gamma -> prod` as you might like.

> This can snowball changes which might cause some confusion if some of those changes were manual changes that result in a `LOCKED` installation. We'll still deliver emails about those as before though, and you'll be able to see the locked state in the `/clusters` view.

Reviewer: ```suggestion
This can snowball changes which might cause some confusion if some of those changes were manual changes that result in a `LOCKED` installation. We’ll still deliver emails about those as before, and you’ll be able to see the locked state in the `/clusters` view.
```
